packages/client/src/runtime/getPrismaClient.ts

```diff
@@ -1060,34 +950,30 @@ new PrismaClient({
         // it went bad, then we rollback the transaction
         await this._engine.transaction('rollback', info).catch(() => {})
 
+        e.clientVersion = this._clientVersion
         throw e // silent rollback, throw original error
       }
 
       return result
     }
 
     /**
-     * Execute a batch of requests in a transaction
-     * @param requests
-     * @param options
+     * Execute queries within a transaction
+     * @param input a callback or a query list
+     * @param options to set timeouts (callback)
+     * @returns
      */
-    private async _transactionWithRequests(
-      requests: Array<Promise<unknown>>,
-      options?: { maxWait: number; timeout: number },
-    ) {
-      return this._transactionWithCallback(async (prisma) => {
-        const transactionId: string = prisma[TX_ID] // some proxy magic
-
-        const _requests = requests.map((request) => {
-          return new Promise((resolve, reject) => {
-            // each request has already been called with `prisma.<call>`
-            // so we inject `transactionId` by intercepting that promise
-            ;(request as any).then(resolve, reject, transactionId)
-          })
-        })
+    async $transaction(input: any, options?: any) {
+      // TODO: remove this once interactive tx became GA
+      if (!this._hasPreviewFlag('interactiveTransactions')) {
```

Comment:
Since preview flags are case-insensitive, I would be curious to know if this condition is still triggered with for example `InTeRaCtiveTransactionS`

Reply:
It works :stuck_out_tongue: